packages/cli/commands/sandbox/create.js

```diff
@@ -27,26 +47,221 @@ exports.handler = async options => {
+      if (usage['DEVELOPER'].available === 0) {
+        const devSandboxLimit = usage['DEVELOPER'].limit;
+        const plural = devSandboxLimit !== 1;
+        const hasDevelopmentSandboxes = getHasSandboxesByType(
+          accountConfig,
+          DEVELOPER_SANDBOX
+        );
+        if (hasDevelopmentSandboxes) {
+          logger.error(
+            i18n(
+              `cli.lib.sandbox.create.failure.alreadyInConfig.developer.${
+                plural ? 'other' : 'one'
+              }`,
+              {
+                accountName: accountConfig.name || accountId,
+                limit: devSandboxLimit,
+              }
+            )
+          );
+        } else {
+          const baseUrl = getHubSpotWebsiteOrigin(
+            getValidEnv(getEnv(accountId))
+          );
+          logger.error(
+            i18n(
+              `cli.lib.sandbox.create.failure.limit.developer.${
+                plural ? 'other' : 'one'
+              }`,
+              {
+                accountName: accountConfig.name || accountId,
+                limit: devSandboxLimit,
+                link: `${baseUrl}/sandboxes-developer/${accountId}/development`,
+              }
+            )
+          );
+        }
+        trackCommandUsage('sandbox-create', { successful: false }, accountId);
+        process.exit(EXIT_CODES.ERROR);
+      }
+    }
+    if (sandboxType === STANDARD_SANDBOX) {
+      if (usage['STANDARD'].available === 0) {
+        const standardSandboxLimit = usage['STANDARD'].limit;
+        const plural = standardSandboxLimit !== 1;
+        const hasStandardSandboxes = getHasSandboxesByType(
+          accountConfig,
+          STANDARD_SANDBOX
+        );
+        if (hasStandardSandboxes) {
+          logger.error(
+            i18n(
+              `cli.lib.sandbox.create.failure.alreadyInConfig.standard.${
+                plural ? 'other' : 'one'
+              }`,
+              {
+                accountName: accountConfig.name || accountId,
+                limit: standardSandboxLimit,
+              }
+            )
+          );
+        } else {
+          const baseUrl = getHubSpotWebsiteOrigin(
+            getValidEnv(getEnv(accountId))
+          );
+          logger.error(
+            i18n(
+              `cli.lib.sandbox.create.failure.limit.standard.${
+                plural ? 'other' : 'one'
+              }`,
+              {
+                accountName: accountConfig.name || accountId,
+                limit: standardSandboxLimit,
+                link: `${baseUrl}/sandboxes-developer/${accountId}/standard`,
+              }
+            )
+          );
+        }
+        trackCommandUsage('sandbox-create', { successful: false }, accountId);
+        process.exit(EXIT_CODES.ERROR);
+      }
+    }
+  } catch (err) {
+    logErrorInstance(err);
+    trackCommandUsage('sandbox-create', { successful: false }, accountId);
+    process.exit(EXIT_CODES.ERROR);
+  }
```

Comment:
We fetch usage limits for sandboxes, and if a portal has no available sandboxes (as per selected type), we exit early and display the following error:
<img width="758" alt="Screenshot 2023-04-26 at 13 28 22" src="https://user-images.githubusercontent.com/16788677/234655658-1964b225-9a0c-4080-bf89-377f65a4936e.png">
<img width="767" alt="Screenshot 2023-04-26 at 13 28 35" src="https://user-images.githubusercontent.com/16788677/234655664-e857b9df-8250-4c6b-b316-a8b56588642b.png">